lib/overcommit/hook/shared/rake_target.rb

```diff
@@ -0,0 +1,24 @@
+module Overcommit::Hook::Shared
+  # runs specified rake targets. It fails on the first non-
+  # successfull exit.
+  #
+  module RakeTarget
+    def run(targets = config['targets'])
+      if targets.nil?
+        raise "RakeTarget: targets parameter is nil. Add at least one task to \
+          the targets parameter. Valid: Array of target names"
+      elsif !targets.is_a?(Array)
+        raise "RakeTarget: targets parameter is not an array. Valid: Array of \
+          rake target names"
+      end
+
+      targets.each do |task|
+        result = execute(command << task)
```

Comment:
Fortunately the `Overcommit::Hook::Base#command` always returns a new array, but I don't believe we use this style of appending a single item using `<<` elsewhere in the code where performance isn't key (preferring a more functional approach), since it can potentially cause problems _if_ the method were to return a reference to a memoized value, causing each invocation to modify the list of command arguments.

Would prefer if we changed it to:

```ruby
  ... = execute(command + [task])
```

...as the overhead of creating a new array in this case isn't worth worrying about.